velox/dwio/common/DirectDecoder.h

```diff
@@ -155,6 +155,7 @@ class DirectDecoder : public IntDecoder<isSigned> {
 
     int32_t numValues = 0;
     auto rows = visitor.rows();
+    auto numValuesBeforePage = visitor.reader().numScanned();
```

Comment:
How hard is it to pass it as an extra parameter of this method?

Reply:
Do you suggest putting `ReadWithVisitorParams` in `facebook::velox::dwio::common` namespace or `facebook::velox::parquet` namespace? If it is put in `facebook::velox::dwio::common` namespace, I think it should be no problem, but it will return to the previous state, that is, the data format without paging needs to pass `ReadWithVisitorParams` when call `facebook::velox::dwio::common::DirectDecoder::fastPath`. If `ReadWithVisitorParams` is put in `facebook::velox::parquet` namespace, how can I get the `ReadWithVisitorParams` of parquet in `facebook::velox::dwio::common::DirectDecoder::fastPath`? I don't have any good ideas at the moment. @Yuhta Do you have any suggestions?

Reply:
You can add an extra parameter `numScanned` to `DirectDecoder::readWithVisitor`.  Pass in 0 whenever this value is not needed.

Reply:
This code has been completely refactored, see https://github.com/facebookincubator/velox/pull/10468/commits/1a633bd889898f1cd82801b7727eb161bf5acb4b

Reply:
Can you remove `numScanned` from column visitors?

Reply:
There is no `numScanned` member variable in `ColumnVisitors.h`